src/proto/single/mod.rs

```diff
@@ -206,19 +208,39 @@ impl JobBuilder {
     }
 }
 
+/// Details on a job's failure.
 #[derive(Serialize, Deserialize, Debug, Clone)]
+#[non_exhaustive]
 pub struct Failure {
-    retry_count: usize,
-    failed_at: String,
+    /// [`Number`](Job::retry) of times this job can be retried.
+    pub retry_count: usize,
+
+    /// Number of remaining retry attempts.
+    #[serde(rename = "remaining")]
+    pub retry_remaining: usize,
+
+    /// Last time this job failed.
+    pub failed_at: DateTime<Utc>,
     #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
This and the other `#[serde]` attributes seem misplaced now — they should come after the doc comments for the relevant field.

Reply:
Sorry for this, must have tilted